pkg/odo/util/validation.go

```diff
@@ -22,3 +24,82 @@ func ValidateName(name string) error {
 	return nil
 
 }
+
+func validateNameAsValidator(name interface{}) error {
+	s := name.(string)
+	return ValidateName(s)
+}
+
+const (
+	// NameValidatorKey is the key used to identify the validator associated with ValidateName function
+	NameValidatorKey           = "odo_name_validator"
+	defaultIntegerValidatorKey = "odo_default_integer"
+)
+
+// Validatable represents a common ancestor for validatable parameters
+type Validatable struct {
+	Required             bool
+	Type                 string
+	AdditionalValidators []string
+}
+
+// AsValidatable allows avoiding type casts in client code
+func (v Validatable) AsValidatable() Validatable {
+	return v
+}
+
+// Validator is a function that validates that the provided interface is conform to expectations or return an error
+type Validator func(interface{}) error
+
+// always validates
+var nilValidator = func(ans interface{}) error { return nil }
+
+var validators = make(map[string]Validator)
```

Comment:
Can we a comment on what this does?

Reply:
Removed it with new commit as this didn't work very well. Trying to explain what it was doing to add a comment made me realize that it was awkward… :)

Reply:
Cool, I'll check again